sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubConsumerClient/EventHubConsumerClientLiveTests.cs

```diff
@@ -771,6 +771,338 @@ await foreach (var partitionEvent in consumer.ReadEventsFromPartitionAsync(parti
+                        {
+                            break;
+                        }
+                    }
+
+                    // Verify the results.
+
+                    Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The publishing and reading should have completed normally.");
+                    Assert.That(readCount, Is.EqualTo(expectedCount), "There was an incorrect number of events received.");
+                    Assert.That(consecutiveEmpties, Is.LessThan(maximumConsecutiveEmpties), "Reading should have ended because the events were all read.");
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubConsumerClient" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ConsumerCanReadFromAllPartitionsWhenUsingAutomaticRouting()
+        {
+            await using (EventHubScope scope = await EventHubScope.CreateAsync(4))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                using var cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromSeconds(90));
+
+                await using (var consumer = new EventHubConsumerClient(EventHubConsumerClient.DefaultConsumerGroupName, connectionString))
+                {
+                    var partitions = await consumer.GetPartitionIdsAsync(cancellationSource.Token);
+                    var eventsPerPartition = 5;
+                    var expectedCount = (eventsPerPartition * partitions.Length);
+                    var readCount = 0;
+                    var consecutiveEmpties = 0;
+                    var maximumConsecutiveEmpties = 10;
+
+                    // Send events without influencing the partition.  Routing is controlled by the service and partitions may not
+                    // receive an even distribution.  Because reading begins at the beginning of the partition by default, these
+                    // should be observed without publishing in the read loop.
+
+                    await using (var producer = new EventHubProducerClient(connectionString))
+                    {
+                        foreach (var partition in partitions)
+                        {
+                            using var batch = await producer.CreateBatchAsync(cancellationSource.Token);
+
+                            for (var index = 0; index < eventsPerPartition; ++index)
+                            {
+                                batch.TryAdd(new EventData(Encoding.UTF8.GetBytes($"Event: { index } for Partition: { partition }")));
+                            }
+
+                            await producer.SendAsync(batch, cancellationSource.Token);
+                        }
```

Comment:
I was using it to denote the context that we created it in, since everything was driven by "eventsPerPartition".    As you mentioned, it doesn't really matter - we could just do a raw count.    I'll convert it over just to avoid any confusion.

Reply:
Decided to stick with the approach of using the partitions to drive the batch size;  converted the event text over to avoid mentioning the partition and avoiding confusion.